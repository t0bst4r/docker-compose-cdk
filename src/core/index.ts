export * from './app';
export * from './project';
export * from './resource';
export * from './stage';
export * from './synth-metadata';
