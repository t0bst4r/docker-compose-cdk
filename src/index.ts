export * from './core';
export * from './naming';
export * from './resources';
