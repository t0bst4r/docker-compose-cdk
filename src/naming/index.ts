export * from './default-naming-strategy';
export * from './naming-strategy';
