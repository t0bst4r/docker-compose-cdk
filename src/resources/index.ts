export * from './config';
export * from './network';
export * from './service';
export * from './service-props';
export * from './volume';
