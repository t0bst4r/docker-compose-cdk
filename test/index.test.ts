import { Construct } from 'constructs';
import { App, DockerImage, Network, PortMapping, Project, RestartPolicy, Service, Volume } from '../src';

test('Multiple Compose Projects', () => {
  const app = new App();

  const firstProject = new Project(app, 'First');
  new Nginx(firstProject, 'Nginx');

  const secondProject = new Project(app, 'Second');
  new Wordpress(secondProject, 'Wordpress');

  expect(app.synth()).toMatchSnapshot();
});


class Nginx extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    const network = new Network(this, 'Network');
    const volume = new Volume(this, 'HtmlData');
    new Service(this, 'Default', {
      image: DockerImage.hub('nginx', 'alpine'),
      restart: RestartPolicy.UNLESS_STOPPED,
      volumes: [
        { source: volume, target: '/usr/share/nginx/html' },
      ],
      ports: [
        PortMapping.http(),
        PortMapping.https(),
      ],
      networks: [{ network }],
    });
  }
}

class Wordpress extends Construct {

  constructor(scope: Construct, id: string) {
    super(scope, id);

    const databaseName = 'exampledb';
    const databaseUser = 'exampleuser';
    const databasePassword = 'examplepass';

    const network = new Network(this, 'Network');

    const dbData = new Volume(this, 'DatabaseData');
    const db = new Service(this, 'DatabaseService', {
      image: DockerImage.hub('mysql', '8.0'),
      restart: RestartPolicy.UNLESS_STOPPED,
      environment: {
        MYSQL_DATABASE: databaseName,
        MYSQL_USER: databaseUser,
        MYSQL_PASSWORD: databasePassword,
        MYSQL_RANDOM_ROOT_PASSWORD: '1',
      },
      volumes: [
        { source: dbData, target: '/var/lib/mysql' },
      ],
      networks: [{ network }],
    });

    const wordpressData = new Volume(this, 'WordpressData');
    new Service(this, 'Default', {
      image: DockerImage.hub('wordpress'),
      restart: RestartPolicy.UNLESS_STOPPED,
      environment: {
        WORDPRESS_DB_HOST: db.dnsNames[0],
        WORDPRESS_DB_NAME: databaseName,
        WORDPRESS_DB_USER: databaseUser,
        WORDPRESS_DB_PASSWORD: databasePassword,
      },
      volumes: [
        { source: wordpressData, target: '/var/www/html' },
      ],
      networks: [{ network }],
    });
  }

}
