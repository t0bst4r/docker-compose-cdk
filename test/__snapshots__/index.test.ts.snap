// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Compose Projects 1`] = `
{
  "projects": [
    {
      "compose": {
        "name": "first",
        "networks": {
          "nginx_network": {
            "driver": "bridge",
            "driver_opts": undefined,
            "ipam": undefined,
          },
        },
        "services": {
          "nginx": {
            "command": undefined,
            "configs": undefined,
            "deploy": {
              "mode": "replicated",
              "replicas": 1,
            },
            "devices": undefined,
            "environment": undefined,
            "image": "nginx:alpine",
            "networks": {
              "nginx_network": {
                "ipv4_address": undefined,
              },
            },
            "ports": [
              "80:80",
              "443:443",
            ],
            "privileged": false,
            "profiles": undefined,
            "restart": "unless-stopped",
            "user": undefined,
            "volumes": [
              {
                "read_only": undefined,
                "source": "nginx_htmldata",
                "target": "/usr/share/nginx/html",
                "type": "volume",
              },
            ],
          },
        },
        "volumes": {
          "nginx_htmldata": {},
        },
      },
      "composeHash": "fecfd64745042024",
      "name": "first",
    },
    {
      "compose": {
        "name": "second",
        "networks": {
          "wordpress_network": {
            "driver": "bridge",
            "driver_opts": undefined,
            "ipam": undefined,
          },
        },
        "services": {
          "wordpress": {
            "command": undefined,
            "configs": undefined,
            "deploy": {
              "mode": "replicated",
              "replicas": 1,
            },
            "devices": undefined,
            "environment": {
              "WORDPRESS_DB_HOST": "second-wordpress-databaseservice-1",
              "WORDPRESS_DB_NAME": "exampledb",
              "WORDPRESS_DB_PASSWORD": "examplepass",
              "WORDPRESS_DB_USER": "exampleuser",
            },
            "image": "wordpress:latest",
            "networks": {
              "wordpress_network": {
                "ipv4_address": undefined,
              },
            },
            "ports": undefined,
            "privileged": false,
            "profiles": undefined,
            "restart": "unless-stopped",
            "user": undefined,
            "volumes": [
              {
                "read_only": undefined,
                "source": "wordpress_wordpressdata",
                "target": "/var/www/html",
                "type": "volume",
              },
            ],
          },
          "wordpress_databaseservice": {
            "command": undefined,
            "configs": undefined,
            "deploy": {
              "mode": "replicated",
              "replicas": 1,
            },
            "devices": undefined,
            "environment": {
              "MYSQL_DATABASE": "exampledb",
              "MYSQL_PASSWORD": "examplepass",
              "MYSQL_RANDOM_ROOT_PASSWORD": "1",
              "MYSQL_USER": "exampleuser",
            },
            "image": "mysql:8.0",
            "networks": {
              "wordpress_network": {
                "ipv4_address": undefined,
              },
            },
            "ports": undefined,
            "privileged": false,
            "profiles": undefined,
            "restart": "unless-stopped",
            "user": undefined,
            "volumes": [
              {
                "read_only": undefined,
                "source": "wordpress_databasedata",
                "target": "/var/lib/mysql",
                "type": "volume",
              },
            ],
          },
        },
        "volumes": {
          "wordpress_databasedata": {},
          "wordpress_wordpressdata": {},
        },
      },
      "composeHash": "a9266a4abc31b34c",
      "name": "second",
    },
  ],
}
`;
